### ISSUE TITLE
Cockpit shadows are weird in certain situations

### ISSUE THREAD

**Issue by gilbertohasnofb** [c172p-team/c172p#1153]
Cockpit shadows are weird in certain situations:

![image](https://user-images.githubusercontent.com/5700795/43370765-c2f0dd1a-937c-11e8-89f4-36f6002038d0.png)

This screenshot was taken at the single parking positions at SAWB. Time was set to afternoon and then I used the time shift to find the exact moment in which the shadows look weird. Seems like a mesh problem? @wlbragg any ideas?

**Comment by gilbertohasnofb** [comment]
Just to be clear, look at the shapes of the shadows, how jagged they are, how they change from bright to dark in places that should be uniform, etc.

**Comment by wlbragg** [comment]
@gilbertohasnofb that has to be the rendering framework and shadow computation at extreme latitudes. I don't think this is anything we did or can fix. It may have to some do with imperfect mesh, but there is nothing I can do about it. Someone should post the criteria to reproduce on the dev list for Thorsten to take a look at maybe.
I noticed it is only one very tight angle with the sun at just the right elevation that this happens on my computer.

**Comment by gilbertohasnofb** [comment]
> shadow computation at extreme latitudes. 

do you mean that the airport location affects that? This airport is in Antarctica and I use it often for testing since since the scenery is lightweight.

> I noticed it is only one very tight angle with the sun at just the right elevation that this happens on my computer.

Yeah, it's a very specific angle that causes it. I am closing this then and I will drop a message to Thorsten.

**Comment by wkitty42** [comment]
On 07/29/2018 09:51 PM, Gilberto Agostinho wrote:
>     shadow computation at extreme latitudes.
> 
> do you mean that the airport location affects that? This airport is in 
> Antarctica and I use it often for testing since since the scenery is lightweight.


you should try Diego Garcia ;)


>     I noticed it is only one very tight angle with the sun at just the right
>     elevation that this happens on my computer.
> 
> Yeah, it's a very specific angle that causes it. I am closing this then and I 
> will drop a message to Thorsten.


try DG and see if you have the issue... it is closer to the equator and should 
provide more info if this is a lighting problem if needed...

**Comment by gilbertohasnofb** [comment]
@wkitty42 Thanks for the tips, I will try those airports and see how they compare.

**Comment by wlbragg** [comment]
@gilbertohasnofb I am seeing this same issue at my local airport KEQA here in Kansas. So It something new. So I guess we really need to make sure this isn't something new that we have introduced. I need to check c172p in fgdata and our master. Also should check other aircraft with shadows such as the 182.

**Comment by wlbragg** [comment]
@gilbertohasnofb this is also in fgdata. I do think it has something to do with the imperfect mesh, but it's not something I can really fix. I think it is just "hard" shadows because of the angles. I also don't think it is something that can be fixed in the effects framework so we may just have to let it go.

**Comment by gilbertohasnofb** [comment]
is there an easy way in Blender to check if some selected points have the same z coordinate? I think most of the points in the panel should be on a plane and these shadows make me think that they are so ever slightly off this plane. But we can also let this go, it's minor.

**Comment by wlbragg** [comment]
I looked at that closely and actually moved all the X coords to the same value all around the recesses on the right side. It didn't help, The panel really needs to be completely retopologized.
This is what the mesh looks like after I fixed the x points to all be the same. As you can see it is still a mess.
This also has to do with using the smoothing tool in Blender. But that is not the only issue and alone cannot fix this.

![panelface2](https://user-images.githubusercontent.com/11069680/43429127-65adbef6-9427-11e8-826b-dab43e399286.jpg)

![panelface](https://user-images.githubusercontent.com/11069680/43429085-2bd40cb2-9427-11e8-826c-192afeb7493d.jpg)

**Comment by gilbertohasnofb** [comment]
Yes, it does look quite messy. I will test and see if I can spot the difference. What strikes me is how non uniform some of the mesh is, e.g. the largest circles around these two gauge positions. We can possibly leave this issue opened in case someone in the future feels like improving this.
